README.md

```diff
@@ -35,95 +36,115 @@ You will need to provide necessary credentials to interact with the Stella Now p
 
 Replace **<Your-API-Key>**, **<Your-API-Secret>**, **<Your-Client-ID>**, **<Your-Organization-ID>**, and **<Your-Project-ID>** with your respective Stella Now credentials.
 
+<div class="alert alert-warning">
+<strong>Note:</strong> Please note that the ClientId used in the StellaNowConfig needs to be unique per connection. If two connections use the same ClientId, then the first connection will be dropped. Always ensure that the ClientId is unique for each connection your application makes.
+</div>
+
 ## Sample Application
-Here is a simple application that uses StellaNowSDK to send user login messages to the Stella Now platform:
+Here is a simple application that uses StellaNowSDK to send user login messages to the Stella Now platform.
+
+This function is the main entry point for our demonstration.
+The **RunAsync** function does a few things:
+
+* It establishes a connection to StellaNow.
+* It creates a series of **UserLoginMessage** messages and sends them using the StellaNow SDK.
+* It checks and logs whether there are any messages left in the queue.
+* Finally, it disconnects from StellaNow and disposes of the services.
 
-    internal class Program
+
+    private async Task RunAsync()
     {
-    private static IServiceProvider _serviceProvider;
-    
-        static async Task Main(string[] args)
-        {
-            RegisterServices();
-    
-            var stellaSdk = _serviceProvider.GetRequiredService<IStellaNowSdk>();
-            var logger = _serviceProvider.GetRequiredService<ILogger<Program>>();
-    
-            // Establish connection
-            await stellaSdk.StartAsync();
-    
-            // Ensure connection is established
-            await Task.Delay(TimeSpan.FromSeconds(5));
-    
-            // Send 5 messages
-            for (int i = 0; i < 5; i++)
-            {
-                var uuid = Guid.NewGuid().ToString();
-    
-                var message = new UserLoginMessage(
-                    uuid, uuid,
-                    DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ")
-                );
-    
-                // Send the message
-                stellaSdk.SendMessage(message);
-                logger.LogInformation("New Message Queued");
-    
-                // Delay between messages
-                await Task.Delay(TimeSpan.FromSeconds(1));
-            }
-    
-            // Disconnect and terminate
-            await stellaSdk.StopAsync();
-            DisposeServices();
-        }
-    
-        private static void RegisterServices()
+        // Establish connection
+        await _stellaSdk.StartAsync();
```

Comment:
adding ful blocks of code into documentation files might easily get outdated, or miss bug fix if we only change the sources

you can change this to reference code bocks directly from sources: https://www.stevemar.net/github-code-in-readme/

Reply:
This documentation is also uploaded as part of the NuGet deployment. The reference blocks do not work there as far as I am concerned.

Reply:
crap, have you tried or there is some article explaining limitation? maintaining both places manually is error-prone